thorlcr/activities/keyedjoin/thkeyedjoinslave.cpp

```diff
@@ -159,20 +253,92 @@ class CJoinGroup : public CSimpleInterfaceOf<IInterface>
                 }
                 ++cur;
             }
+            while (--numRowArrays);
             ReleaseRoxieRow(rowArrays);
             rowArrays = nullptr;
-            totalRows = 0;
+            totalRhsSz = 0;
+        }
+    }
+    void freeRowsAndStream()
+    {
+        ::Release(rowStream);
+        rowStream = nullptr;
+        freeRows();
+    }
+
+    inline const void *_getNextRhsClear(offset_t &fpos, JoinGroupRhsState &rhsState)
+    {
+        while (rhsState.arr != (rowArrays+numRowArrays)) // end of array marker
+        {
+            if (rhsState.arr->rows)
+            {
+                while (rhsState.pos < rhsState.arr->numRows)
+                {
+                    Row &row = rhsState.arr->rows[rhsState.pos++];
+                    if (row.rhs)
+                    {
+                        fpos = row.fpos;
```

Comment:
There is lots of code to preserve filepos.  I'm not at all sure it is ever used, and adds lots of complication.  I will check to see if it would ever cause an issue to not allow access to it.

Reply:
Isn't filepos needed for full keyed joins?

Reply:
yes, unless there's something I don't know.

But you're right, it's a lot of effort and a waste of time and complexity in cases where it's not.
There's a old comment somewhere, saying it would be good to know if used or not.

I can't reduce complexity much if can be used (full keyed join), but if I know it's not used (half-keyed join), I can avoid quite a lot of overhead, requesting, transmitting and tracking etc.